splink/linker.py

```diff
@@ -1174,6 +1175,18 @@ def find_matches_to_new_records(
         else:
             new_records_tablename = records_or_tablename
 
+        # If our df_concat_with_tf table already exists, use backwards induction to
+        # find  all underlying term frequency tables.
+        if self._table_exists_in_database("__splink__df_concat_with_tf"):
+            sqls = compute_term_frequencies_from_concat_with_tf(self)
+
+            for sql in sqls:
+                self._enqueue_sql(sql["sql"], sql["output_table_name"])
+        else:
+            # This queues up our cols_with_tf and df_concat_with_tf tables.
+            # self._initialise_df_concat_with_tf(materialise=False)
+            self._initialise_df_concat_with_tf(materialise=False)
```

Comment:
Do we think there are many instances in which a user would call `find_matches_to_new_records` a single time?

If it's going to be called multiple times, it may be better to always compute the term frequency tables the first time?

Reply:
Do you think we should cache _**all**_ term frequency tables, or just `concat_with_tf`?

Caching _**all**_ tables has the potential to generate a large number of random tables for users, with limited performance improvements.

If we initialise `concat_with_tf` (which I do agree we should probably do given the above), it's simply a matter of running a quick `distinct()` call on specific columns.